crates/core/src/types/engine/payload.rs

```diff
@@ -0,0 +1,158 @@
+    /// B) `LegacyTransaction` (An rlp encoded LegacyTransaction)
+    fn decode(&self) -> Result<Transaction, RLPDecodeError> {
+        // Look at the first byte to check if it corresponds to a TransactionType
+        match self.0.first() {
+            // First byte is a valid TransactionType
+            Some(tx_type) if *tx_type < 0x7f => {
+                // Decode tx based on type
+                let tx_bytes = &self.0.as_ref()[1..];
+                match *tx_type {
+                    // Legacy
+                    0x0 => {
+                        LegacyTransaction::decode_rlp(tx_bytes).map(Transaction::LegacyTransaction)
+                    } // TODO: check if this is a real case scenario
+                    // EIP1559
+                    0x2 => EIP1559Transaction::decode_rlp(tx_bytes)
+                        .map(Transaction::EIP1559Transaction),
+                    _ => unimplemented!("We don't know this tx type yet"),
+                }
+            }
+            // LegacyTransaction
+            _ => LegacyTransaction::decode_rlp(self.0.as_ref()).map(Transaction::LegacyTransaction),
+        }
+    }
+}
+
+// TODO: Consider adding a struct Block { block_header, block_body } and implementing TryInto
+#[allow(unused)]
+impl ExecutionPayloadV3 {
+    fn into_block(self) -> Result<(BlockHeader, BlockBody), RLPDecodeError> {
+        let block_body = BlockBody {
+            transactions: self
+                .transactions
+                .iter()
+                .map(|encoded_tx| encoded_tx.decode())
+                .collect::<Result<Vec<_>, RLPDecodeError>>()?,
+            ommers: vec![],
+            withdrawals: self.withdrawals,
+        };
+        Ok((
+            BlockHeader {
+                parent_hash: self.parent_hash,
+                ommers_hash: H256::zero(), // TODO: Use appropiate value
+                coinbase: self.fee_recipient,
+                state_root: self.state_root,
+                transactions_root: block_body.compute_transactions_root(),
+                receipt_root: self.receipts_root,
+                logs_bloom: self.logs_bloom.into(),
+                difficulty: 0.into(),
+                number: self.block_number,
+                gas_limit: self.gas_limit,
+                gas_used: self.gas_used,
+                timestamp: self.timestamp,
+                extra_data: self.extra_data,
+                prev_randao: self.prev_randao,
+                nonce: 0, // TODO: Use appropiate value
+                base_fee_per_gas: self.base_fee_per_gas,
+                withdrawals_root: H256::zero(), // TODO: Use result of root calculation once implemented
+                blob_gas_used: self.blob_gas_used,
+                excess_blob_gas: self.excess_blob_gas,
+                parent_beacon_block_root: H256::zero(), // TODO: Use appropiate value
```

Comment:
This is received in the same RPC call as the `ExecutionPayloadV3`: https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#engine_newpayloadv3